src/index.d.ts

```diff
@@ -1,7 +1,8 @@
+type Function = (...args: any[]) => any;
 /**
  * Factory as defined in the injectable property: either a function with eventually a single named dependencies argument object or a value
  */
-export type FactoryFn<FactoryLike> = FactoryLike extends (args: any) => any
+export type FactoryFn<FactoryLike> = FactoryLike extends Function
```

Comment:
it is not quite the same. Factories have only one argument

Reply:
Yes, i took an unnecessary shortcut by sharing a generic `Function` type. I removed it from this PR.